src/components/PageBlocks/RecordListFilter.vue

```diff
@@ -0,0 +1,249 @@
+<template>
+  <div>
+    <b-btn
+      variant="link p-0 ml-1"
+      @click="onOpen"
+      v-b-tooltip.focus
+      :id="selectedFieldName"
+      class="btn-filter"
+      v-bind:class="{ 'text-primary': inFilter }"
+    >
+      <font-awesome-icon :icon="['fas', 'filter']" />
+    </b-btn>
+    <b-tooltip
+      custom-class="ctooltip"
+      ref="tooltip"
+      class="ctooltip1"
+      :target="selectedFieldName"
+      triggers="focus"
+    >
+      <div class="filter-tooltip">
+        <section
+          class="filter-row"
+          v-for="(filter, index) in componentFilter"
+          :key="index"
+        >
+          <label v-if="index === 0">Where</label>
+          <b-form-select
+            v-else
+            v-model="filter.condition"
+            :options="conditions"
+            size="md"
+          ></b-form-select>
+          <b-form-select
+            v-model="filter.name"
+            :options="fields"
+            value-field="key"
+            text-field="label"
+            size="md"
+            @change="onChange(filter.name, index)"
+          ></b-form-select>
+          <b-form-select
+            v-model="filter.equation"
+            :options="equations"
+            size="md"
+          ></b-form-select>
+
+          <!-- Multi field type, String,Number,Bool etc -->
```

Comment:
You can replace lines 47-72 with a single field-editor component, make sure you properly define a mock like we talked about. We can discuss this further on our sync

Reply:
Initial implementation to be done on call